src/stdlib/base.rs

```diff
@@ -79,4 +79,47 @@ pub fn load_base<'gc>(mc: MutationContext<'gc, '_>, root: LuaRoot<'gc>, env: Tab
         }),
     )
     .unwrap();
+
+    env.set(
+        mc,
+        String::new_static(b"type"),
+        Callback::new_immediate(mc, |args| {
+            if args.len() == 0 {
+                return Err(RuntimeError(Value::String(String::new_static(
+                    b"Missing argument to type",
+                )))
+                .into());
+            }
+            match args.get(0).cloned().unwrap() {
+                // Lua5.3 does not differentiate between a floating point and integer number
+                Value::Integer(_) => Ok(CallbackResult::Return(vec![Value::String(
+                    String::new_static(b"number"),
+                )])),
+                a => Ok(CallbackResult::Return(vec![Value::String(
+                    String::new_static(a.type_name().as_bytes()),
+                )])),
+            }
+        }),
+    )
+    .unwrap();
+
+    env.set(
+        mc,
+        String::new_static(b"select"),
+        Callback::new_immediate(mc, |args| {
+            match args.get(0).cloned().unwrap_or(Value::Nil).to_integer() {
+                Some(n) if n >= 1 && (n as usize) <= args.len() => Ok(CallbackResult::Return(
+                    args[n as usize..args.len()].to_vec(),
+                )),
```

Comment:
I don't think this is right, isn't this off by one?  n == 1 will miss the first argument?

Reply:
Nope, because the arguments to select are `select(number_to_select, other_args...)` so n == 1 will give the arguments from index 1 (the first of `other_args`) to the end.

Reply:
oh right, duh